### ISSUE TITLE
Error with Compile KinectFusion with cmake 

### ISSUE THREAD

**Issue by LuanTran07** [yuxng/DA-RNN#13]
Could you help me about the problem with " No package 'nanoflann' found" ? 
Could you support to adding module 'nanoflann' ?
Thank you very much!

**Comment by lizhihuit** [comment]
Do you have solved this problem,I meet the same issue. @LuanTran07 @yuxng

**Comment by kevinkit** [comment]
I faced the same problem, have yo useen #10 ? There is shown how to install nanoflann (basically downlad it, compile it, and use make install). 


However, be aware that the repo still faces some memory leaks during execution.

If you have set it up, and you can get it running without the memory leaks, feel free to share your advice

**Comment by weiguochow** [comment]
when i have installed the module nanaflann,but when i make in Kinect_fusion,the error as follows:
8 errors detected in the compilation of "/tmp/tmpxft_0000463a_00000000-7_nonrigidIcp.cpp1.ii".
# --error 0x2 --
CMake Error at kfusion_generated_nonrigidIcp.cu.o.cmake:266 (message):
  Error generating file
  /home/alex/lizhi/DA-RNN-master/lib/kinect_fusion/build/CMakeFiles/kfusion.dir/src/optimization/./kfusion_generated_nonrigidIcp.cu.o


CMakeFiles/kinectFusion.dir/build.make:112: recipe for target 'CMakeFiles/kfusion.dir/src/optimization/kfusion_generated_nonrigidIcp.cu.o' failed
make[2]: *** [CMakeFiles/kfusion.dir/src/optimization/kfusion_generated_nonrigidIcp.cu.o] Error 1
CMakeFiles/Makefile2:67: recipe for target 'CMakeFiles/kinectFusion.dir/all' failed
make[1]: *** [CMakeFiles/kinectFusion.dir/all] Error 2
Makefile:83: recipe for target 'all' failed
make: *** [all] Error 2
@kevinkit @

**Comment by kevinkit** [comment]
@weiguochow I guess you mean, that you have installed nanoflann  ? (you wrote nanaflann, i just hope that there is no confusion and an other framework was isntalled accidently).


I am not sure if the mentioned error comes from nanoflann, can you make sure that it is the fault of nanoflann? If not, maybe open another Issue.


However, while we are at it, you must give a little bit more detail about the error code, and at what point he fails. Make sure to enable all the outputs provided by CMake

**Comment by weiguochow** [comment]
@kevinkit  thx,when i install another version noflann,I have another problem as follows:
/home/alex/lizhi/DA-RNN-master/lib/kinect_fusion/include/df/util/tupleHelpers.h(104): warning: integer conversion resulted in a change of sign

/home/alex/lizhi/DA-RNN-master/lib/kinect_fusion/include/df/transform/rigid.h(14): error: qualified name is not allowed

/home/alex/lizhi/DA-RNN-master/lib/kinect_fusion/include/df/transform/rigid.h(14): error: explicit type is missing ("int" assumed)

/home/alex/lizhi/DA-RNN-master/lib/kinect_fusion/include/df/transform/rigid.h(14): error: expected a ";"

/home/alex/lizhi/DA-RNN-master/lib/kinect_fusion/include/df/transform/rigid.h(24): error: identifier "Transform" is undefined

/home/alex/lizhi/DA-RNN-master/lib/kinect_fusion/include/df/transform/rigid.h(28): error: identifier "Transform" is undefined

/home/alex/lizhi/DA-RNN-master/lib/kinect_fusion/include/df/transform/rigid.h(32): error: identifier "Transform" is undefined

/home/alex/lizhi/DA-RNN-master/lib/kinect_fusion/include/df/transform/rigid.h(36): error: identifier "Transform" is undefined

/home/alex/lizhi/DA-RNN-master/lib/kinect_fusion/include/df/transform/rigid.h(52): error: identifier "Transform" is undefined

8 errors detected in the compilation of "/tmp/tmpxft_00005af5_00000000-7_marchingCubes.cpp1.ii".
# --error 0x2 --
CMake Error at kfusion_generated_marchingCubes.cu.o.cmake:266 (message):
  Error generating file
  /home/alex/lizhi/DA-RNN-master/lib/kinect_fusion/build/CMakeFiles/kfusion.dir/src/surface/./kfusion_generated_marchingCubes.cu.o


CMakeFiles/kinectFusion.dir/build.make:84: recipe for target 'CMakeFiles/kfusion.dir/src/surface/kfusion_generated_marchingCubes.cu.o' failed
make[2]: *** [CMakeFiles/kfusion.dir/src/surface/kfusion_generated_marchingCubes.cu.o] Error 1
CMakeFiles/Makefile2:67: recipe for target 'CMakeFiles/kinectFusion.dir/all' failed
make[1]: *** [CMakeFiles/kinectFusion.dir/all] Error 2
Makefile:83: recipe for target 'all' failed
make: *** [all] Error 2

**Comment by weiguochow** [comment]
@kevinkit My email is 1983587450@qq.com,can you send me the package of nanoflann or the Kinect_Fusion,thank you

**Comment by kevinkit** [comment]
https://github.com/jlblancoc/nanoflann/releases , the newest version should work fine. 

However I am quite unsure at the moment and will post more details once I am in my office again (tomorrow)

**Comment by lizhihuit** [comment]
@kevinkit 
can you give me the package of the compiled DA-RNN, because I have to do some expriments now, please,thank you

**Comment by weiguochow** [comment]
thx,I just install the newest version of nanoflann,and the problem is shown above,

**Comment by kevinkit** [comment]
Yes as soon as I get back - however, there is still the memory leak  which needs to be fixed. The program will crash right on start.

**Comment by kevinkit** [comment]
Anyhow, have you tried sudo?

**Comment by weiguochow** [comment]
yes ,i used sudo ,but there still have problem,can you give me the code compiled.thank you. weiguochow@gmail.com

**Comment by lizhihuit** [comment]
@kevinkit  Itry my best to install the kinectFusion,but there still have error...I don't know the reason

**Comment by kevinkit** [comment]
We managed to compile it with running make -j32 and right after that running make again (no joke) and then make install.

However anybody who successfully complied it was not able to run the program since it crashed due to the memory leak, as mention in other issues such as #6 .

I think only the author has a real running version

**Comment by lizhihuit** [comment]
OK，thx

发自网易邮箱大师
On 11/02/2017 21:13, Kevin wrote: We managed to compile it with running make -j32 and right after that running make again (no joke) and then make install.
However anybody who successfully complied it was not able to run the program since it crashed due to the memory leak, as mention in other issues such as #6 .
I think only the author has a real running version

—You are receiving this because you commented.Reply to this email directly, view it on GitHub, or mute the thread.


  
  




{"api_version":"1.0","publisher":{"api_key":"05dde50f1d1a384dd78767c55493e4bb","name":"GitHub"},"entity":{"external_key":"github/yuxng/DA-RNN","title":"yuxng/DA-RNN","subtitle":"GitHub repository","main_image_url":"https://cloud.githubusercontent.com/assets/143418/17495839/a5054eac-5d88-11e6-95fc-7290892c7bb5.png","avatar_image_url":"https://cloud.githubusercontent.com/assets/143418/15842166/7c72db34-2c0b-11e6-9aed-b52498112777.png","action":{"name":"Open in GitHub","url":"https://github.com/yuxng/DA-RNN"}},"updates":{"snippets":[{"icon":"PERSON","message":"@kevinkit in #13: We managed to compile it with running make -j32 and right after that running make again (no joke) and then make install.\r\n\r\nHowever anybody who successfully complied it was not able to run the program since it crashed due to the memory leak, as mention in other issues such as #6 .\r\n\r\nI think only the author has a real running version "}],"action":{"name":"View Issue","url":"https://github.com/yuxng/DA-RNN/issues/13#issuecomment-341416741"}}}
